src/vmm/src/vmm_config/machine_config.rs

```diff
@@ -90,13 +90,25 @@ pub struct MachineConfigUpdate {
     )]
     pub smt: Option<bool>,
     /// A CPU template that it is used to filter the CPU features exposed to the guest.
-    #[serde(default, skip_serializing_if = "Option::is_none")]
-    pub cpu_template: Option<StaticCpuTemplate>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        deserialize_with = "some_option"
+    )]
+    pub cpu_template: Option<Option<StaticCpuTemplate>>,
```

Comment:
nit: it might be worth noting why it is wrapped with two `Options` (The outer `Option` indicates whether `cpu_template` field is provided or not, and the inner one indicates whether the given `cpu_template` is "None" or a specific CPU template.)

Reply:
good point!

Reply:
Done :)